packages/core/src/browser/shell/shell-layout-restorer.ts

```diff
@@ -49,6 +49,17 @@ export namespace StatefulWidget {
     }
 }
 
+export interface StoreControlableWidget {
```

Comment:
it would be good to avoid changing core interfaces and keep change local.

Could `TerminalWidget.storeState` returns nothing for pseudo terminal or an empty object? Or an object with one property to indicate that `psedo: true`?

Reply:
> it would be good to avoid changing core interfaces and keep change local.
> 
> Could `TerminalWidget.storeState` returns nothing for pseudo terminal or an empty object? Or an object with one property to indicate that `psedo: true`?

I have thought about return null when storeState, but it also use a bit memory

Reply:
Is that OK？

Reply:
yes, it is fine, it is tiny compare to all terminal options.